cmake/check_cuda.cmake

```diff
@@ -30,7 +36,33 @@ if(USE_CUDA AND CMAKE_CUDA_COMPILER)
   # enable_language(CUDA)
   # message(STATUS "CMAKE_CUDA_COMPILER_VERSION: ${CMAKE_CUDA_COMPILER_VERSION}")
   # set(CUDA_PROPAGATE_HOST_FLAGS ON)
+
+  if(WIN32 AND NOT CMAKE_CUDA_FLAGS_INIT)
+      set(CMAKE_CUDA_FLAGS "${CMAKE_CUDA_FLAGS} /DWIN32 /D_WINDOWS /DWINAPI_FAMILY=100 /DWINVER=0x0A00 /D_WIN32_WINNT=0x0A00 /DNTDDI_VERSION=0x0A000000 
```

Comment:
For a local build, you do not need the flags.

Reply:
If you set it('/Qspectre'), people who didn't install the VC Qspectre runtime will fail to build our code. 
Therefore I think it's better to not having it.

Reply:
I thought BinSkim requires `/QSpectre` and is widely used in https://github.com/microsoft/onnxruntime-genai/blob/main/cmake/presets/CMakeWinConfigPresets.json

TLDR: We already need VC QSpectre runtime to build our code.

Reply:
You can build this repo without the preset files.